source/common/modules/markdown-editor/hooks/codeblock-classes.js

```diff
@@ -64,6 +79,7 @@ function applyCodeblockClasses (cm) {
     // codeblocks themselves should not be styled)
     if (codeBlockRE.test(line)) {
       isCodeBlock = !isCodeBlock
+      codeBlockLines.push(i)
```

Comment:
Interesting solution ... but if I understand it correctly, this array doesn't include all `codeBlockLines` but only those at a switch from codeblock to markdown, right? I was thinking that we can make use of the toggle of the `isCodeBlock`-variable above and just add a new variable "was previous not code block?" ... right?

Reply:
> this array doesn't include all `codeBlockLines` but only those at a switch from codeblock to markdown, right?

Yes, only the lines that match the pre-existing `codeBlockRE`.

> I was thinking that we can make use of the toggle of the isCodeBlock-variable above and just add a new variable

Since we need to look behind and ahead, I think that would be overly complex. I mean, you may save 2 bytes of memory per code block, but you'll have to detect that a boolean that is true was false in the previous round so you need to store the boolean in a new boolean at the end of the looped block, but there are cases where a `continue` is used so you cannot rely on end of loop execution, but maybe you can if you wrote the code and know the logic, but for a random contributor it's surprisingly hard to navigate the small routine as it is. Especially since `isCodeBlock` starts as `false` so `isCodeBlock = !isCodeBlock` is ambiguous and should be written as `isCodeBlock = true` but maybe not because perhaps a toggle was intended in some unfinished logic for the indented code block.

Sorry, I'm rambling. This is my attempt at an elegant solution that touches the least amount of code. If you prefer a boolean approach, could you show how you'd do it? The patch is minor enough to ignore this and create a new PR, although this PR allows edits by maintainers too.